plugins/Kaleidoscope-LEDControl/README.md

```diff
@@ -1,6 +1,118 @@
 # Kaleidoscope-LEDControl
 
 This is a plugin for [Kaleidoscope][fw], for controlling the LEDs, and LED
-effects.
+effects. It is also a building block for plugins that control LEDs.
 
  [fw]: https://github.com/keyboardio/Kaleidoscope
+
+## Using the extension
+
+## Plugin methods
+
+### `.next_mode(void)`
+
+> Activates the next LED mode. Cycles to the first LED mode if the current LED
+> mode is the last one.
+
+### `.prev_mode(void)`
+
+> Activates the previous LED mode. Cycles to the last LED mode if the current
+> LED mode is the first one.
+
+### `.get_mode()`
+
+
+### `.set_mode(uint8_t mode_id)`
+
+> Activates a LED mode by its index in the firmware. If the index exceeds the
+> numer of led modes, the method returns early.
+
+### `.get_mode_index()`
+
+> Returns the index of the currently active LED mode.
+
+### `.refreshAll()`
+
+> If the hardware has LEDs and LEDs are enabled, turn all LEDs off and then
+> trigger the current LED mode to refresh.
+
+### `.setCrgbAt(uint8_t led_index, cRGB crgb)`
+
+> Sets the specified LED to the provided color.
+
+### `.setCrgbAt(KeyAddr key_addr, cRGB color)`
+
+> Sets the LED for the specified key to the provided color.
+
+### `.getCrgbAt(uint8_t led_index)`
+
+> Get the LED color of the specified LED.
+
+### `.getCrgbAt(KeyAddr key_addr)`
+
+> Get the LED color of the LED for the specified key.
+
+### `.syncLeds(void)`
```

Comment:
This one seems pretty straight forward to me, but I failed to come up with a good way to phrase this.